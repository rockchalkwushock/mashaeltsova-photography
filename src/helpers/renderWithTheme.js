import React from 'react'
import { mount } from 'enzyme'
import { ThemeProvider } from 'styled-components'
import theme from '../utils/theme'

/**
 * @fileOverview
 *
 * HOC for testing components with `theme` prop being used.
 */

const renderWithTheme = Component =>
  mount(<ThemeProvider theme={theme}>{Component}</ThemeProvider>)

export default renderWithTheme
