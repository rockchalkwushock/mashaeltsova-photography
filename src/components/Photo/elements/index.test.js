import React from 'react'

import StyledPhoto from './'
import { renderWithTheme } from '../../../helpers'

const photo = {
  sizes: {
    base64: '/static/pic.jpg',
    height: 200,
    src: '/static/pic.jpg',
    srcSet: '/static/pic.jpg',
    srcWebp: '/static/pic.jpg',
    srcSetWebp: '/static/pic.jpg',
    width: 496
  }
}

describe('Styled Component: <StyledPhoto />', () => {
  let wrapper
  beforeAll(() => {
    wrapper = renderWithTheme(
      <StyledPhoto alt="pic" sizes={photo.sizes} title="pic" />
    )
  })
  test('should render markup', () => {
    expect(wrapper).toMatchSnapshot()
  })
  // REVIEW: Fails to see styles???
  test.skip('should have styles', () => {
    expect(wrapper).toHaveStyleRule('border', 'none')
    expect(wrapper).toHaveStyleRule(
      'box-shadow',
      '2px 2px 10px 1px rgba(0, 0, 0, 1)'
    )
  })
})
