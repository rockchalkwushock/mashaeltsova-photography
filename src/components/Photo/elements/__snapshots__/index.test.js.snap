// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styled Component: <StyledPhoto /> should render markup 1`] = `
.c0 {
  border: none;
  box-shadow: 2px 2px 10px 1px rgba(0,0,0,1);
}

<ThemeProvider
  theme={
    Object {
      "accent": "#df744a",
      "bg": "#ffffff",
      "fg": "#bfd8d2",
      "highlight": "#dcb239",
      "lowlight": "#36454f",
      "sizes": Object {
        "lg": "900px",
        "md": "736px",
        "sm": "540px",
      },
    }
  }
>
  <StyledPhoto
    alt="pic"
    sizes={
      Object {
        "base64": "/static/pic.jpg",
        "height": 200,
        "src": "/static/pic.jpg",
        "srcSet": "/static/pic.jpg",
        "srcSetWebp": "/static/pic.jpg",
        "srcWebp": "/static/pic.jpg",
        "width": 496,
      }
    }
    title="pic"
  >
    <UnstyledPhoto
      alt="pic"
      className="c0"
      sizes={
        Object {
          "base64": "/static/pic.jpg",
          "height": 200,
          "src": "/static/pic.jpg",
          "srcSet": "/static/pic.jpg",
          "srcSetWebp": "/static/pic.jpg",
          "srcWebp": "/static/pic.jpg",
          "width": 496,
        }
      }
      title="pic"
    >
      <Image
        Tag="div"
        alt="pic"
        className="c0"
        fadeIn={true}
        sizes={
          Object {
            "base64": "/static/pic.jpg",
            "height": 200,
            "src": "/static/pic.jpg",
            "srcSet": "/static/pic.jpg",
            "srcSetWebp": "/static/pic.jpg",
            "srcWebp": "/static/pic.jpg",
            "width": 496,
          }
        }
        title="pic"
      >
        <div
          className=" gatsby-image-outer-wrapper"
          style={
            Object {
              "position": "relative",
            }
          }
        >
          <div
            className="c0 gatsby-image-wrapper"
            style={
              Object {
                "overflow": "hidden",
                "position": "relative",
              }
            }
          >
            <div
              style={
                Object {
                  "paddingBottom": "NaN%",
                  "width": "100%",
                }
              }
            />
            <Img
              alt="pic"
              src="/static/pic.jpg"
              style={
                Object {
                  "opacity": 0,
                  "transitionDelay": "0.25s",
                }
              }
              title="pic"
            >
              <img
                alt="pic"
                src="/static/pic.jpg"
                style={
                  Object {
                    "height": "100%",
                    "left": 0,
                    "objectFit": "cover",
                    "objectPosition": "center",
                    "opacity": 0,
                    "position": "absolute",
                    "top": 0,
                    "transition": "opacity 0.5s",
                    "transitionDelay": "0.25s",
                    "width": "100%",
                  }
                }
                title="pic"
              />
            </Img>
            <Img
              alt="pic"
              onLoad={[Function]}
              src="/static/pic.jpg"
              srcSet="/static/pic.jpg"
              style={
                Object {
                  "opacity": 1,
                }
              }
              title="pic"
            >
              <img
                alt="pic"
                onLoad={[Function]}
                src="/static/pic.jpg"
                srcSet="/static/pic.jpg"
                style={
                  Object {
                    "height": "100%",
                    "left": 0,
                    "objectFit": "cover",
                    "objectPosition": "center",
                    "opacity": 1,
                    "position": "absolute",
                    "top": 0,
                    "transition": "opacity 0.5s",
                    "width": "100%",
                  }
                }
                title="pic"
              />
            </Img>
            <noscript
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<img width=\\"496\\" height=\\"200\\" src=\\"/static/pic.jpg\\" srcset=\\"/static/pic.jpg\\" alt=\\"pic\\" title=\\"pic\\" sizes=\\"\\" style=\\"position:absolute;top:0;left:0;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/>",
                }
              }
            />
          </div>
        </div>
      </Image>
    </UnstyledPhoto>
  </StyledPhoto>
</ThemeProvider>
`;
